### PR TITLE
docs: client libraries outline

### PR DESCRIPTION
**Pull request by nicecui**
## What's Changed in this PR

new client library docs

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/v0.5/en/user-guide/client-libraries/java.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+---
+template: template.md
+---
+# Java
+
```
Comment (nicecui) [inline docs/v0.5/en/user-guide/client-libraries/java.md:5]: @fengjiachun Please add the content of Java refer to Go's documentation.

File: docs/v0.5/en/user-guide/client-libraries/template.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+
```
Comment (nicecui) [inline docs/v0.5/en/user-guide/client-libraries/template.md:1]: PTAL from here

File: docs/v0.5/en/summary.yml [hunk 3]
```diff
@@ -37,34 +37,22 @@
...
-    - SDK-Libraries:
-      - go
-      - java
-      # - erlang
-      # - javaScript
   - query-external-data
+  - Client-Libraries:
+    - overview
+    - go
+    - java
```
Comment (killme2008) [inline docs/v0.5/en/summary.yml:55]: We can add erlang， and i'll write the manual.

**Review by killme2008 — APPROVED** [review]
I am ok with the new outline, let's do it!

File: docs/v0.5/en/user-guide/client-libraries/go.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+```
+
+}
+
+{template query-library-raw-sql%
+
+```go
+type Monitor struct {
+	ID          int64     `gorm:"primaryKey"`
```
[the remark this answers is not in the document]
Comment (yuanbohan) [reply docs/v0.5/en/user-guide/client-libraries/go.md:270]: any advice?

[the remark this answers is not in the document]
Comment (nicecui) [reply docs/v0.5/en/user-guide/client-libraries/go.md:270]: Can you confirm that the following struct works correctly?

```go
type Monitor struct {
	Host        string    `gorm:"primaryKey"`
	Cpu         float64  
	Ts          time.Time `gorm:"primaryKey"`
}
```

**Review by sunng87 — APPROVED** [review]

**Pull request merged**